cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -371,7 +371,6 @@ private ProgramOptions createProgramOptions(CommandLine cmdLine, TwillContext co
     Map<String, String> arguments = Maps.newHashMap(original.getArguments().asMap());
     arguments.put(ProgramOptionConstants.INSTANCE_ID, Integer.toString(context.getInstanceId()));
     arguments.put(ProgramOptionConstants.INSTANCES, Integer.toString(context.getInstanceCount()));
-    arguments.put(ProgramOptionConstants.RUN_ID, original.getArguments().getOption(ProgramOptionConstants.RUN_ID));
```

Comment:
It's already inside the `original.getArguments()` (line 371 above), hence not need to set it again.